src/lava/lib/optimization/solvers/generic/read_gate/process.py

```diff
@@ -53,7 +60,12 @@ def __init__(
         self.target_cost = Var(shape=(1,), init=target_cost)
         self.best_solution = Var(shape=shape, init=-1)
         for id in range(num_in_ports):
-            setattr(self, f"cost_in_{id}", InPort(shape=(1,)))

```

Comment:
I think the comments in the next lines are already on the docstring?

Reply:
If you don't mind, I would still like to keep them here. Makes understanding easier for people who go through the code, since the way how we split 32bit into uint24+sint8 is not straightforward.